db/migrate/20240913075420_add_index_to_users_email.rb

```diff
@@ -0,0 +1,5 @@
+class AddIndexToUsersEmail < ActiveRecord::Migration[6.1]
+  def change
+    add_index :users, :email, unique: true
```

Comment:
e tìm hiểu index trong sql nhé:
- Định nghĩa
- Công dụng
- Dùng index trong những trường hợp nào
- Ưu và nhược điểm của index

Reply:
- index là cấu trúc dữ liệu được dùng để cải thiện tốc độ truy vấn dữ liệu trong db giúp tìm nhanh thông tin cụ thể mà ko cần phải quét toàn bộ bảng
- công dụng: giảm thời gian tìm kiếm dữ liệu và các thao tác xử lý dữ liệu như sắp xếp, nhóm dữ liệu 
- thường dùng index trong các trường hợp như tìm kiếm, sắp xếp, nhóm dữ liệu và được tạo tự động trên các cột khoá chính và khoá ngoại
- ưu điểm: tăng tốc độ truy vấn, sắp xếp, nhóm dữ liệu và hỗ trợ khoá chính, khoá ngoại
- nhược điểm: tăng kích thước dữ liệu, tăng thời gian cập nhật và tốn tài nguyên